modulemd/v2/modulemd-translation-entry.c

```diff
@@ -0,0 +1,623 @@
+  object_class->get_property = modulemd_translation_entry_get_property;
+  object_class->set_property = modulemd_translation_entry_set_property;
+
+  properties[PROP_LOCALE] = g_param_spec_string (
+    "locale",
+    "Locale",
+    "The locale for this translation entry. It must correspond to the format "
+    "specified by libc locale names. This field may only be set on object "
+    "construction and is immutable afterwards.",
+    TE_DEFAULT_STRING,
+    G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS | G_PARAM_CONSTRUCT_ONLY);
+
+  properties[PROP_SUMMARY] =
+    g_param_spec_string ("summary",
+                         "Summary",
+                         "The summary of this module stream translated into "
+                         "the language specified by locale.",
+                         TE_DEFAULT_STRING,
+                         G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS);
+
+  properties[PROP_DESCRIPTION] =
+    g_param_spec_string ("description",
+                         "Description",
+                         "The description of this module stream translated "
+                         "into the language specified by locale.",
+                         TE_DEFAULT_STRING,
+                         G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS);
+
+  g_object_class_install_properties (object_class, N_PROPS, properties);
+}
+
+
+static void
+modulemd_translation_entry_init (ModulemdTranslationEntry *self)
+{
+  self->profile_descriptions =
+    g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_free);
+}
+
+
+/* === YAML Functions === */
+
+static void *
+modulemd_translation_entry_parse_yaml_profiles (yaml_parser_t *parser,
+                                                ModulemdTranslationEntry *te,
+                                                GError **error)
+{
+  MMD_INIT_YAML_EVENT (event);
+  gboolean done = FALSE;
+  gboolean in_map = FALSE;
+  gchar *profile_name = NULL;
+  gchar *value = NULL;
+
+  g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
+
+  /* Parse the profiles */
+  while (!done)
+    {
+      YAML_PARSER_PARSE_WITH_EXIT (parser, &event, error);
+
+      switch (event.type)
+        {
+        case YAML_MAPPING_START_EVENT: in_map = TRUE; break;
+
+        case YAML_MAPPING_END_EVENT:
+          in_map = FALSE;
+          done = TRUE;
+          break;
+
+        case YAML_SCALAR_EVENT:
+          if (!in_map)
+            {
+              MMD_YAML_ERROR_EVENT_EXIT (
+                error, event, "Missing mapping in translation entry profiles");
+              break;
+            }
+
+          profile_name = g_strdup ((const gchar *)event.data.scalar.value);
+          value = modulemd_yaml_parse_string (parser, error);
+          g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
+          modulemd_translation_entry_set_profile_description (
+            te, profile_name, value);
+          g_free (profile_name);
```

Comment:
```suggestion
          g_clear_pointer(&profile_name, g_free);
```

This is preferred to avoid double-free when the autoptr cleanup hits.